pkg/trait/service.go

```diff
@@ -61,6 +61,15 @@ func (t *serviceTrait) Configure(e *Environment) (bool, error) {
 		return false, nil
 	}
 
+	// in case the knative-service and service trait are enabled, the knative-service has priority
+	// then this service is disabled
+	if e.GetTrait(knativeServiceTraitID) != nil {
```

Comment:
Have you considered any other place where to provide this check? I mean, it would be nice not to have dependency between traits (if it is possible, of course).

Reply:
This PR is only a short-term solution. In long-term we should refactor the service traits and how they are configured and theirs dependency to each other. 

The idea would be to configure all service traits with one single trait config and have a new `service.type` trait config property that selects the service nature (default vs. knative vs. any other)